modules/kargoBidAdapter.js

```diff
@@ -4,286 +4,477 @@ import { registerBidder } from '../src/adapters/bidderFactory.js';
-      method: 'GET',
-      url: `${HOST}/api/v2/bid`,
-      data: `json=${encodedParams}`,
-      currency: currency
-    });
-  },
-  interpretResponse: function(response, bidRequest) {
-    let bids = response.body;
-    const bidResponses = [];
-    for (let bidId in bids) {
-      let adUnit = bids[bidId];
-      let meta = {
-        mediaType: BANNER
-      };
-
-      if (adUnit.metadata && adUnit.metadata.landingPageDomain) {
-        meta.clickUrl = adUnit.metadata.landingPageDomain[0];
-        meta.advertiserDomains = adUnit.metadata.landingPageDomain;
+  if (socialCanvas != null) {
+    krakenParams.socan = socialCanvas
+  }
+
+  // User Agent Client Hints / SUA
+  const uaClientHints = deepAccess(firstBidRequest, REQUEST_KEYS.SUA);
+  if (uaClientHints) {
+    const suaValidAttributes = []
+
+    for (const suaKey of SUA_ATTRIBUTES) {
+      const suaValue = uaClientHints[suaKey]
+      if(!suaValue) {
+        continue
       }
 
-      if (adUnit.mediaType && SUPPORTED_MEDIA_TYPES.includes(adUnit.mediaType)) {
-        meta.mediaType = adUnit.mediaType;
+      // Do not pass any empty strings
+      if(typeof suaValue == 'string' && suaValue.trim() === '') {
+        continue
       }
 
-      const bidResponse = {
-        ad: adUnit.adm,
-        requestId: bidId,
-        cpm: Number(adUnit.cpm),
-        width: adUnit.width,
-        height: adUnit.height,
-        ttl: 300,
-        creativeId: adUnit.id,
-        dealId: adUnit.targetingCustom,
-        netRevenue: true,
-        currency: adUnit.currency || bidRequest.currency,
-        mediaType: meta.mediaType,
-        meta: meta
-      };
-
-      if (meta.mediaType == VIDEO) {
-        bidResponse.vastXml = adUnit.adm;
+      switch (suaKey) {
+        case SUA.MOBILE, SUA.SOURCE:
+          if (suaValue < 1) {
+            return;
+          };
+        default:
+          suaValidAttributes.push(suaKey);
       }
+    }
+
+    krakenParams.device.sua = pick(uaClientHints, suaValidAttributes);
+  }
+
+  const validPageId = getLocalStorageSafely(CERBERUS.PAGE_VIEW_ID) != null 
+  const validPageTimestamp = getLocalStorageSafely(CERBERUS.PAGE_VIEW_TIMESTAMP) != null
+  const validPageUrl = getLocalStorageSafely(CERBERUS.PAGE_VIEW_URL) != null
+
+
+  const page = {}
+  if(validPageId) {
+    page.id = getLocalStorageSafely(CERBERUS.PAGE_VIEW_ID)
+  }
+  if(validPageTimestamp) {
+    page.timestamp = Number(getLocalStorageSafely(CERBERUS.PAGE_VIEW_TIMESTAMP))
+  }
+  if(validPageUrl) {
+    page.url = getLocalStorageSafely(CERBERUS.PAGE_VIEW_URL)
+  }
+  if (!_.isEmpty(page)) {
+    krakenParams.page = page
+  }
+  
+  return Object.assign({}, bidderRequest, {
+    method: BIDDER.REQUEST_METHOD,
+    url: `${BIDDER.HOST}${BIDDER.REQUEST_ENDPOINT}`,
+    data: krakenParams,
+    currency: currency
+  });
+}
+
+function interpretResponse(response, bidRequest) {
+  let bids = response.body;
+  const bidResponses = [];
+  for (let bidID in bids) {
```

Comment:
Consider using `Object.keys().forEach()` or `Object.entries()` over a for in loop for some performance gain and modernizing this to use ES6.